packages/core/database/lib/connection.js

```diff
@@ -3,11 +3,19 @@
 'use strict';
 
 const knex = require('knex');
+const path = require('path');
 
-const SqliteClient = require('knex/lib/dialects/sqlite3/index');
-
-const trySqlitePackage = (packageName) => {
+const tryResolvePackage = (packageName) => {
   try {
+    // if this is within a strapi project, make sure it has this driver listed in project dependencies
+    const packageJSON = require(path.resolve('.', 'package.json'));
+    if (
+      packageJSON?.strapi?.uuid &&
+      !Object.keys(packageJSON?.dependencies).includes(packageName)
+    ) {
+      return false;
+    }
```

Comment:
The goal is to only load packages that are listed in the project's package.json so that a cached module doesn't get used when it's not explicitly requested, but I've realized this might actually be really bad to do, since it sort of makes createConnection aware of the project.

The only thing I can think of is to move the package check outside of createConnection and instead tell createConnection which package to use. Will go back to the drawing board and see what I can come up with...